overlord/managers_test.go

```diff
@@ -2408,20 +2417,30 @@ func (ms *mgrsSuite) testUpdateWithAutoconnectRetry(c *C, updateSnapName, remove
 	// force hold state on first removal task to hit Retry error
 	ts2.Tasks()[0].SetStatus(state.HoldStatus)
 
-	st.Unlock()
-	err = ms.o.Settle(3 * time.Second)
-	st.Lock()
-	c.Assert(err, NotNil)
-	c.Assert(err, ErrorMatches, `Settle is not converging`)
-
+	// Settle is not converging here because of the task in Hold status, therefore
+	// it always hits given timeout before we carry on with the test. We're
+	// interested in hitting the retry condition on auto-connect task, so
+	// instead of passing a generous timeout to Settle(), repeat Settle() a number
+	// of times with an aggressive timeout and break as soon as we reach the desired
+	// state of auto-connect task.
 	var retryCheck bool
-	for _, t := range st.Tasks() {
-		if t.Kind() == "auto-connect" {
-			c.Assert(strings.Join(t.Log(), ""), Matches, `.*Waiting for conflicting change in progress...`)
-			retryCheck = true
+	var autoconnectLog string
+	for i := 0; i < 2000 && !retryCheck; i++ {
```

Comment:
this is clunky but seems to work and I don't have a better idea, OTOH 2000 confused me and seems accessive, here it finishes in a couple of loops. 10 or 50 would seem less prone to make readers worry